src/groupByKey.ts

```diff
@@ -2,6 +2,19 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T extends object>(
+  items: T[],
+  key: keyof T,
+): GroupsMap<T> {
+  return items.reduce((acc, item) => {
+    const keyValue = item[key] as keyof GroupsMap<T>;
```

Comment:
Avoid using `as` in TS
```suggestion
    const keyValue = String(item[key]);
```